chain/stmgr/call.go

```diff
@@ -219,7 +224,12 @@ func (sm *StateManager) CallWithGas(ctx context.Context, msg *types.Message, pri
 		}
 	}
 
-	fromActor, err := vmi.StateTree().GetActor(msg.From)
+	stTree, err := sm.StateTree(state)
+	if err != nil {
+		return nil, err
+	}
+
+	fromActor, err := stTree.GetActor(msg.From)
```

Comment:
So, this is going to fail if the actor was changed by one of the messages in `priorMsgs`. We have a few options:

1. Flush to get the current root CID of the state. Unfortunately, that'll flush things back to badger which we don't want to do.
2. Flush, but wrap the blockstore in a buffered blockstore.
3. Expose a way to call get-actor on the FVM through the FFI.

Reply:
@Stebalien Nice catch! I'm tempted to go with 2, unless you think the FVM being able to support `GetActor` is independently valuable.

Reply:
I do think being able to access the FVM state while running is useful, but I'd punt on that for now and do the "easy" thing.

Reply:
But state access will likely involve:

- A `GetActor` call (address -> actor)
- A `ResolveActor` call (address -> id)
- A "blockstore" export. The reverse of the CGO blockstore.